core/lib/zksync_core/src/lib.rs

```diff
@@ -396,17 +396,19 @@ pub async fn initialize_components(
     let (prometheus_health_check, prometheus_health_updater) =
         ReactiveHealthCheck::new("prometheus_exporter");
     app_health.insert_component(prometheus_health_check);
-    let prometheus_task = prom_config.run(stop_receiver.clone());
+    let pt_stop_receiver = stop_receiver.clone();
+    let prometheus_task = prom_config.run(pt_stop_receiver);
     let prometheus_task = tokio::spawn(async move {
         prometheus_health_updater.update(HealthStatus::Ready.into());
         let res = prometheus_task.await;
         drop(prometheus_health_updater);
         res
     });
 
+    let cb_stop_receiver = stop_receiver.clone();
     let mut task_futures: Vec<JoinHandle<anyhow::Result<()>>> = vec![
         prometheus_task,
-        tokio::spawn(circuit_breaker_checker.run(cb_sender, stop_receiver.clone())),
+        tokio::spawn(async move { circuit_breaker_checker.run(cb_stop_receiver).await }),
```

Comment:
`async move { x.await }` can be replaced by simply `x` (and then you don't need to create `cb_stop_receiver`, the `stop_receiver.clone()` will work (since there will be no `move`).